addons/base_import_module/models/base_import_module.py

```diff
@@ -1,49 +1,46 @@
+# -*- coding: utf-8 -*-
 import base64
 from StringIO import StringIO
 from io import BytesIO
-from openerp.osv import osv, fields
+from openerp import api, fields, models
 
-class base_import_module(osv.TransientModel):
+
+class BaseImportModule(models.TransientModel):
     """ Import Module """
     _name = "base.import.module"
     _description = "Import Module"
 
-    _columns = {
-        'module_file': fields.binary('Module .ZIP file', required=True),
-        'state':fields.selection([('init','init'),('done','done')], 'Status', readonly=True),
-        'import_message': fields.char('Import message'),
-        'force': fields.boolean('Force init', help="Force init mode even if installed. (will update `noupdate='1'` records)"),
-    }
-
-    _defaults = {
-        'state': 'init',
-        'force': False,
-    }
+    module_file = fields.Binary('Module .ZIP file', required=True)
+    state = fields.Selection([('init','init'),('done','done')], 'Status', readonly=True, default='init')
+    import_message = fields.Char()
+    force = fields.Boolean(string='Force init', help="Force init mode even if installed. (will update `noupdate='1'` records)")
 
-    def import_module(self, cr, uid, ids, context=None):
-        module_obj = self.pool.get('ir.module.module')
-        data = self.browse(cr, uid, ids[0] , context=context)
+    @api.multi
+    def import_module(self):
+        module_obj = self.env['ir.module.module']
+        data = self.browse(self.ids[0])
         zip_data = base64.decodestring(data.module_file)
         fp = BytesIO()
         fp.write(zip_data)
-        res = module_obj.import_zipfile(cr, uid, fp, force=data.force, context=context)
-        self.write(cr, uid, ids, {'state': 'done', 'import_message': res[0]}, context=context)
-        context = dict(context, module_name=res[1])
+        res = module_obj.import_zipfile(fp, force=data.force)
+        self.browse(self.ids).write({'state': 'done', 'import_message': res[0]})
```

Comment:
Gets data from ids[0] but updates all the ids, is that really normal? Maybe it would make sense to just write on the already-browsed `data`?